### DISCUSSION TITLE
*Noob*   Bluetooth classic works great, can't figure out BLE

### DISCUSSION THREAD

**Discussion by GneeChee** [mKeRix/room-assistant#778] [Q&A]
New to all of this so please be gentle :)

I've got a working setup for one room using bluetooth classic.   I think I read somewhere that BLE can be a bit more accurate, so I'm trying to switch to that.   However, I can't get the config right, I guess.    I keep finding different configs that people say are working for them, but none of them work for me.   I've read the docs, but in all honesty - most of it is beyond my grasp.  

Anyway - here is my config.   I have 2 mac addresses because my watch says that its bluetooth mac is one thing, but _hcitool lescan_ reports it as a different mac.   So, I put in both :)    One, the other, or both - none of it works for me.

Any help would be greatly appreciated.   Also, I'm more than happy to supply logs ... but I have no clue where to find them.

(sorry, but I can't figure out how to paste in the config file and keep the formatting/spacing)

***
***EDIT: ***
Not sure if this is my problem, nor how to correct it, but I ran "room-assistant -v", and ran across this message -- 
  "warn - ConfigService: No configuration found in /home/pi/room-assistant/config/config".   I don't understand why it's looking in config/config, but I created that subdirectory and copied local.yml to there.   

Restarted RA, then got this --
"info - BluetoothLowEnergyService: Discovered new BLE peripheral Galaxy Watch (D002) LE with ID d0fe41d12004 and RSSI -64", which I assume is good, right?  It's not the MAC address the watch gives, but the MAC that hcitool lescan reports.

However, the automation I had set up for Bluetooth Classic doesn't trigger.   I realize I'm missing something (obvious), but I can't seem to find it.
***

***local.yml***
  weight: 2
  autoDiscovery: true
  port: 6425
global:
  instanceName: MyBedroom
  integrations:
    - homeAssistant
    - bluetoothLowEnergy
homeAssistant:
  mqttUrl: 'mqtt://192.168.xxx.xxx:1883'
  mqttOptions:
    username: 'mqtt'
    password: 'xxxxxxxx'
bluetoothLowEnergy:
  timeout: 60
  allowlist:
    - d0fe41d12004
    - d07fa0e81002
  tagOverrides:
    d0fe41d12004:
      name: Richard's Watch
    d07fa0e81002:
      name: Richard's Other Watch

**Comment by GneeChee** [comment]
Okay, so I just proved my Noobishness.    The automations were configured improperly.   The above config appears to work just fine.   However, I did remove the second watch MAC address.    Question - on the Watch, do Bluetooth Classic and BLE use separate MAC addresses?   Samsung states that the watch supports BLE, but it never mentions a different MAC.   Curious.

So now you have my permission to officially whack me 1000 times with a wet noodle.   Sorry to have bothered you guys :)

**Reply by mKeRix** [reply]
Bluetooth Classic and Low Energy can run from the same chip, so they will indeed use the same MAC address. BLE also offers some privacy-focussed features, which are often times implemented on these wearable devices. One of these is that the MAC address can be switched to something seemingly random when advertising the device, to prevent people tracking you unknowingly. The address isn't actually totally random though, it is encrypted - to be able to decrypt it again the receiving device needs the Identity Resolving Key (IRK) of the device, which is only exchanged after pairing.

Long story short, currently there is no easy way in room-assistant to track such devices with BLE without pairing (and tbh, I haven't tested this either yet!). It's something that I would love to see become easier though, I just haven't figured out how to get these keys in a way that you don't need to pair every Pi to your watch.

**Reply by GneeChee** [reply]
Hey bud,  thanks for the explanation.   I'm new to all of this so obviously I need to do a bit more reading up on it all. 

In the meantime,  I'm positive that you in all of your wizardry can get the pairing issues figured out :)

It's amazing the sheer amount of work we will do just to save a few seconds by automating a light bulb,  eh?

**Comment by Scott8586** [comment]
```
ConfigService: No configuration found in /home/pi/room-assistant/config/config". I don't understand why it's looking in config/config, but I created that subdirectory and copied local.yml to there.
```

room-assistant is sensitive to the current working directory, and looks one directory down, in `config`.  If you are in ~pi/room-assistant/config, when you start it it will look in the ~pi/room-assistant/config/config directory for your configuration.

I'm not sure,  but your cluster info may not be recognized, it should look more like this:


```
global:
  instanceName: MyBedroom
  integrations:
    - homeAssistant
    - bluetoothLowEnergy
cluster:
  weight: 2
  autoDiscovery: true
  port: 6425

```

**Reply by GneeChee** [reply]
Hey thanks for responding.    I figured out the config thing literally moments before I read your post :)

As to the cluster info .. I don't really understand that yet,  but it's next on my list to tackle.   Thanks a bazillion for your input on that!
